56.py

```diff
@@ -0,0 +1,22 @@
+from typing import List
+
+class Solution:
+    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
+        intervals.sort(key=self.compare)        
+        merged = []
+        current = intervals[0]
+        
+        for incoming in intervals[1:]:
```

Comment:
Should give names to `incoming[0]`, `incoming[1]`, `current[0]`, and `current[1]` for readability. For example,

```python
for incoming_start, incoming_end in intervals[1:]

Reply:
How do you know the retrieved list only has two elements in it?

Reply:
What you get each time by iterating through `intervals[1:]` is not a list; what you get is an element from `intervals[1:]`, and each element in the `intervals` list has two fields: `intervals[i][0]` and `intervals[i][1]`. Here I am naming `interval[i][0]` and `interval[i][1]` as `incoming_start` and `incoming_end`, respectively.